src/pages/api/proxy/[...path].ts

```diff
@@ -0,0 +1,53 @@
+import { EventEmitter } from 'events';
+
+import { createProxyMiddleware, fixRequestBody } from 'http-proxy-middleware';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+// Define error messages in a constant object
+const ERROR_MESSAGES = {
+  PROXY_ERROR: 'Proxy error',
+  PROXY_HANDLER_ERROR: 'Proxy handler error',
+};
+
+// Increase the max listeners to avoid memory leak
+EventEmitter.defaultMaxListeners = 20;
+
+const apiProxy = createProxyMiddleware<NextApiRequest, NextApiResponse>({
```

Comment:
Would be great to add a comment here to explain what this file does and why it's needed.